src/system/user.cr

```diff
@@ -0,0 +1,62 @@
+require "crystal/system/user"
+
+class System::User
+  # Raised on user lookup failure.
+  class NotFound < Exception
+  end
+
+  private getter sys_user : Crystal::System::User
+
+  private def initialize(@sys_user)
+  end
+
+  # Returns the user associated with the given name, if it exists.
+  #
+  # Raises `Errno` if a system error occurs.
+  def self.from_name?(name)
+    sys_user = Crystal::System::User.from_name?(name)
+    new(sys_user) if sys_user
+  end
+
+  # Returns the user associated with the given name.
+  #
+  # Raises `NotFound` if no such user exists.
+  # See `from_name?`.
+  def self.from_name(name)
+    self.from_name?(name) || raise NotFound.new("no such user: #{name}")
```

Comment:
```suggestion
    from_name?(name) || raise NotFound.new("No such user: #{name}")
```

- Exception messages should start with uppercase
- using `self` is redundant

ditto elsewhere